dynamicdns.go

```diff
@@ -200,126 +194,98 @@ func (a App) checkIPAndUpdateDNS() {
 	currentIPs = removeDuplicateIPs(currentIPs)
 
 	// do a diff of current and previous IPs to make DNS records to update
-	updatedRecsByZone := make(map[string][]libdns.Record)
 	for _, ip := range currentIPs {
 		for zone, domains := range allDomains {
+			// find all records for the zone
+			recs, err := a.dnsProvider.GetRecords(a.ctx, zone)
```

Comment:
This will `Get` multiple times per `ip`. The previous approach was only fetching records _once_ per update loop, but here it may fetch multiple times.